.github/workflows/rust-build-and-test.yml

```
@@ -99,5 +108,99 @@ jobs:
         echo "PROTOC_INCLUDE=/usr/local/include" >> $GITHUB_ENV
     - name: Build
       run: make build
+    -
+      name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+    -
+      name: Login to Public ECR
+      uses: docker/login-action@v2
+      with:
+        registry: public.ecr.aws
+        username: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        password: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      env:
+        AWS_REGION: us-east-1
+    -
+      name: Build and Publish Docker Image
+      id: build
+      run: |
+        SHA_TAG=$(echo ${{ github.SHA }} | head -c 12)
+        BUILD_TAG=$SHA_TAG
+        COMMIT_MESSAGE=$(git log -1 --pretty=%B)
+        if [[ $(echo "$COMMIT_MESSAGE" | grep 'chore: version v') ]]; then
+          RELEASE_TAG=$(cargo metadata --format-version=1 --no-deps | jq '.packages[0].version' | tr -d '"')
+          # Use the release tag to deploy, if one is available.
+          BUILD_TAG=$RELEASE_TAG
+        fi
+        if [[ "${{ github.event_name }}" == "pull_request" && "${{ github.base_ref }}" == "main" ]]; then
+          # Use PR commit sha on PRs for main
+          RELEASE_TAG=$(echo ${{ github.event.pull_request.head.sha }} | head -c 12)
+          BUILD_TAG=$RELEASE_TAG
+        fi
+        make SHA="${{ github.SHA }}" SHA_TAG="$SHA_TAG" RELEASE_TAG="$RELEASE_TAG" publish-docker
```

I think we can simplify this section:
```suggestion
        BUILD_TAG=$(echo ${{ github.event.pull_request.head.sha }} | head -c 12)
        make SHA_TAG="$BUILD_TAG" publish-docker
```

---

Another thing to note is that `make publish-docker` will override `latest`. Could update `publish.sh` to take additional tags as input and pass `latest` when called from CD and nothing when called from here.